examples/official-site/sqlpage/migrations/23_uploaded_file_functions.sql

```diff
@@ -105,6 +105,108 @@ select ''redirect'' as component,
 where sqlpage.uploaded_file_mime_type(''myfile'') not in (select mime_type from allowed_mime_types);
 ```
 '
+),
+(
+    'uploaded_file_name',
+    '0.23.0',
+    'file-description',
+    'Returns the `filename` value in the `content-disposition` header.
+
+## Example: saving uploaded file metadata for later download
+
+### Making a form
+
+```sql
+select ''form'' as component, ''handle_file_upload.sql'' as action;
+select ''myfile'' as name, ''file'' as type, ''File'' as label;
+```
+
+### Handling the form response
+
+### Inserting an arbitrary file as a [data URL](https://en.wikipedia.org/wiki/Data_URI_scheme) into the database
+
+In `handle_file_upload.sql`, one can process the form results like this:
+
+```sql
+insert into uploaded_files (fname, content, uploaded) values (
+  sqlpage.uploaded_file_name(''myfile''),
+  sqlpage.read_file_as_data_url(sqlpage.uploaded_file_path(''myfile'')),
+  CURRENT_TIMESTAMP
+);
+```
+
+> *Note*: Data URLs are larger than the original file, so it is not recommended to use them for large files.
+
+There is not currently support in SQLPage to download files, but a web server with access to the database
+can provide the download with the original file name.
+
+Here is an example of a simple web server using `bottle` (`python`):
```

Comment:
Why?

Reply:
Because I didn't think of the data uri as an option for download. :)